RadixWallet/Features/SettingsFeature/Preferences/Preferences.swift

```diff
@@ -0,0 +1,130 @@
+		case exportLogsButtonTapped
+		case exportLogsDismissed
+	}
+
+	public enum InternalAction: Sendable, Equatable {
+		case loadedAppPreferences(AppPreferences)
+	}
+
+	public struct Destination: DestinationReducer {
+		public enum State: Sendable, Hashable {
+			case depositGuarantees(DefaultDepositGuarantees.State)
+			case hiddenEntities(AccountAndPersonaHiding.State)
+			case gateways(GatewaySettings.State)
+		}
+
+		public enum Action: Sendable, Equatable {
+			case depositGuarantees(DefaultDepositGuarantees.Action)
+			case hiddenEntities(AccountAndPersonaHiding.Action)
+			case gateways(GatewaySettings.Action)
+		}
+
+		public var body: some ReducerOf<Self> {
+			Scope(state: /State.depositGuarantees, action: /Action.depositGuarantees) {
+				DefaultDepositGuarantees()
+			}
+			Scope(state: /State.hiddenEntities, action: /Action.hiddenEntities) {
+				AccountAndPersonaHiding()
+			}
+			Scope(state: /State.gateways, action: /Action.gateways) {
+				GatewaySettings()
+			}
+		}
+	}
+
+	@Dependency(\.appPreferencesClient) var appPreferencesClient
+
+	public init() {}
+
+	public var body: some ReducerOf<Self> {
+		Reduce(core)
+			.ifLet(destinationPath, action: /Action.destination) {
+				Destination()
+			}
+	}
+
+	private let destinationPath: WritableKeyPath<State, PresentationState<Destination.State>> = \.$destination
+
+	public func reduce(into state: inout State, viewAction: ViewAction) -> Effect<Action> {
+		switch viewAction {
+		case .appeared:
+			return .run { send in
+				let appPreferences = await appPreferencesClient.getPreferences()
+				await send(.internal(.loadedAppPreferences(appPreferences)))
+			}
+
+		case .depositGuaranteesButtonTapped:
+			let depositGuarantee = state.appPreferences?.transaction.defaultDepositGuarantee ?? 1
+			state.destination = .depositGuarantees(.init(depositGuarantee: depositGuarantee))
+			return .none
+
+		case .hiddenEntitiesButtonTapped:
+			state.destination = .hiddenEntities(.init())
+			return .none
+
+		case .gatewaysButtonTapped:
+			state.destination = .gateways(.init())
+			return .none
+
+		case let .developerModeToogled(isEnabled):
+			state.appPreferences?.security.isDeveloperModeEnabled = isEnabled
+			guard let preferences = state.appPreferences else { return .none }
+			return .run { _ in
+				try await appPreferencesClient.updatePreferences(preferences)
+			}
+
+		case .exportLogsButtonTapped:
+			state.exportLogsUrl = Logger.logFilePath
+			return .none
+
+		case .exportLogsDismissed:
+			state.exportLogsUrl = nil
+			return .none
+		}
+	}
+
+	public func reduce(into state: inout State, internalAction: InternalAction) -> Effect<Action> {
+		switch internalAction {
+		case let .loadedAppPreferences(appPreferences):
+			state.appPreferences = appPreferences
+			return .none
+		}
+	}
+
+	public func reduceDismissedDestination(into state: inout State) -> Effect<Action> {
+		if
+			case let .depositGuarantees(depositGuarantees) = state.destination,
```

Comment:
I don't think `if case` should be broken up, and the ending brace does not need a newline. Some ex-Swift developers in the team may disagree with me...